bin/docker-image-tool.sh

```diff
@@ -113,16 +115,26 @@ function build {
     KYUUBI_ROOT="$CTX_DIR/base"
   fi
 
-  # cp spark for kyuubi as submit client
-  # if user set -s(spark-provider), use if
-  # else use builtin spark
+  local BUILD_ARGS=(${BUILD_PARAMS})
+
+  # mkdir spark-binary to cache spark
+  # clean cache if spark-binary exists
   if [[ ! -d "$KYUUBI_ROOT/spark-binary" ]]; then
     mkdir "$KYUUBI_ROOT/spark-binary"
+  else
+    rm -rf "$KYUUBI_ROOT/spark-binary/*"
   fi
-  if [[ ! -d "$SPARK_HOME" ]]; then
-    error "Cannot found dir $SPARK_HOME, you must configure SPARK_HOME correct."
+  # if with spark image means, kyuubi won't copy spark from local.
+  # In this case, we just pass SPARK_HOME without copy.
+  if [[ "${SPARK_PROVIDED}" != "false" ]]; then
+    BUILD_ARGS+=(--build-arg spark_home=$SPARK_HOME)
```

Comment:
A little confuse here. the `SPARK_HOME` means the directory inside docker container. But in general, the `SPARK_HOME` means the directory in real world.

So should the name be `SPARK_HOME_IN_DOCKER` ? and we don't need the flag `SPARK_PROVIDED` any more since we can detect `SPARK_HOME_IN_DOCKER` if existed.

Also we should document the `SPARK_HOME_IN_DOCKER`

Reply:
I will try.